clients/application-kafka/src/main/java/org/eclipse/hono/application/client/kafka/impl/KafkaBasedCommandSender.java

```diff
@@ -157,4 +284,43 @@ public KafkaBasedCommandSender(
 
         return props;
     }
+
+    private Future<Void> subscribeForCommandResponse(final String tenantId,
+            final Handler<DownstreamMessage<KafkaMessageContext>> messageHandler, final Span span) {
+        if (commandResponseSubscriptions.get(tenantId) != null
+                && commandResponseSubscriptions.get(tenantId).one() != null) {
```

Comment:
The `commandResponseSubscriptions.get(tenantId).one() != null` check looks superfluous and rather confusing here.